scripts/synapse_port_db

```diff
@@ -924,17 +1007,24 @@ if __name__ == "__main__":
         },
     }
 
-    postgres_config = yaml.safe_load(args.postgres_config)
+    hs_config = yaml.safe_load(args.postgres_config)
+
+    config = HomeServerConfig()
+    config.parse_config_dict(hs_config, "", "")
 
-    if "database" in postgres_config:
-        postgres_config = postgres_config["database"]
+    if "database" in hs_config:
+        postgres_config = hs_config["database"]
 
-    if "name" not in postgres_config:
-        sys.stderr.write("Malformed database config: no 'name'")
-        sys.exit(2)
-    if postgres_config["name"] != "psycopg2":
-        sys.stderr.write("Database must use 'psycopg2' connector.")
-        sys.exit(3)
+        if "name" not in postgres_config:
```

Comment:
We should either do these checks before using hs_config to instantiate a `HomeserverConfig`, or change them to work on `config` instead. I'd vote for the former, but ymmv.